connector/openshift/openshift.go

```diff
@@ -109,9 +119,30 @@ func (c *Config) Open(id string, logger log.Logger) (conn connector.Connector, e
 		Scopes:      []string{"user:info"},
 		RedirectURL: c.RedirectURI,
 	}
+
+	if err := openshiftConnector.queryOAuth2Endpoints(); err != nil {
+		cancel()
+		return nil, fmt.Errorf("failed to query oauth2 endpoint: %v",
+			err)
+	}
+
 	return &openshiftConnector, nil
 }
 
+func (c *openshiftConnector) queryOAuth2Endpoints() error {
+	for _, endpoint := range []string{c.oauth2Config.Endpoint.AuthURL, c.oauth2Config.Endpoint.TokenURL} {
+		req, err := http.NewRequest(http.MethodGet, endpoint, nil)
+		if err != nil {
+			return err
+		}
+		_, err = c.httpClient.Do(req)
```

Comment:
Note, it is possible that the context logic above was inserted because it is planned to add context-awareness to `Open` in a later version. In this case, pass `ctx` to this method, and change this line to
```suggestion
		_, err = c.httpClient.Do(req.WithContext(ctx))
```
Might be good to do this proactively - it's easy to remove this if `ctx` goes away (which for good hygiene shouldn't happen in this PR anyway), but hard to spot that it's missing if `ctx` is meant to stay.

Reply:
Added the context awareness to the request within #2